src/aspire/source/relion.py

```diff
@@ -163,7 +163,8 @@ def load_single_mrcs(filepath, df):
 
             for future in futures.as_completed(to_do):
                 data_indices, data = future.result()
-                im[:, :, data_indices-start] = data
+                for idx in  data_indices-start:
```

Comment:
I can come back to this, but ostensibly no.  There isn't much way around situations where we've stored things inside out. I tried the usual suspects and the naive loop was fast.  Maybe I've missed one you'd prefer to try...

```
test1 average call time (secs): 0.007777272114995867
test2a average call time (secs): 0.008408336571999825
test2b average call time (secs): 0.008417469299980438
test3a average call time (secs): 0.008867768257972784
test3b average call time (secs): 0.00876319267001236
```

GH wouldn't let me upload the script, so I pasted it here.
```
import timeit
import numpy as np

sz = (128,128,512)

A = np.arange(np.prod(sz),dtype=np.float32).reshape(sz)
B = np.zeros(sz[::-1])

def test1():
    """test loop"""
    # lets mask off half
    for i in range(0, sz[0], 2):
        B[i] = A[:,:,i]

def test2a():
    # lets mask off half
    inds = range(0, sz[0], 2)
    B[inds] = np.moveaxis(A[:,:, inds], -1, 0)

def test2b():
    # lets mask off half
    inds = range(0, sz[0], 2)
    B[inds] = np.moveaxis(A, -1, 0)[inds]

def test3a():
    # lets mask off half,
    # pretend we don't care about img axes for testing purposes
    inds = range(0, sz[0], 2)
    B[inds] = A.T[inds]

def test3b():
    # lets mask off half
    inds = range(0, sz[0], 2)
    B[inds] = A[:,:, inds].T
    
REPS = 1000
TESTS = ('test1', 'test2a', 'test2b', 'test3a', 'test3b')
setup = ''
for test in TESTS:
    setup += f'from __main__ import {test}; '

for test in TESTS:
    t = timeit.timeit(f"{test}()", setup=setup, number=REPS)
    print(f'{test} average call time (secs): {t/REPS}')


```

Reply:
I'm not following completely. Why are we copying every other image in the test script? Is this to circumvent some sort of optimization? Also, does this take into account that `A` is likely column-major in storage (looking at `load_single_mrcs`, it seems likely). I guess I would expect something like `test3a` to perform the best (if we remove this subsampling by two). Ideally, we'd like to perform a straight copy from the MRCS file into memory.

Reply:
I copied every other just to make a dummy mask, since the original code is looping through some arbitrary (to me) indices.  I was just trying to demonstrate that the simple loops performance is fine.  I didn't realize you were implying the underlying data was already transposed via numpy.  For some reason I assumed the data was not tainted yet :).

I see what you mean about `load_single_mrcs`. Cool, I just tried to cancel out both the transposes, test running now.

Reply:
To reproduce original behavior I had to swap the image axis (after wiping out the transpose + stack axis swap).  If you prefer I do not swap the image axis just let me know and I will transpose the reference data instead. Thanks.

Reply:
I see. I'd rather transpose the reference data then. This is likely to become an unnecessary bottleneck otherwise.

Reply:
Sure, after changing the image axes in the test reference, this section resolved to a simple slice assignment...